subsys/bluetooth/mesh/settings.c

```diff
@@ -183,7 +133,92 @@ static void store_pending(struct k_work *work)
 	}
 }
 
+static int mesh_commit(void)
+{
+	if (!bt_mesh_subnet_next(NULL)) {
+		/* Nothing to do since we're not yet provisioned */
+		return 0;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_MESH_PB_GATT)) {
+		bt_mesh_proxy_prov_disable(true);
+	}
+
+	bt_mesh_net_settings_commit();
+	bt_mesh_model_settings_commit();
+
+	atomic_set_bit(bt_mesh.flags, BT_MESH_VALID);
+
+	bt_mesh_start();
+
+	return 0;
+}
+
+static int mesh_export(int (*export_func)(const char *name, const void *val,
+					  size_t val_len))
+{
+	BT_DBG("");
+
+	/* Even if settings_save passes settings_save_one as export_func, it is not
+	 * necessary to be like that, e.g. when h_export isn't called by settings_save.
+	 * In this case, it can be unsafe to use export_func. Since it is not obligatory
+	 * to use export_func when implementing h_export, use settings_save_one directly.
+	 */
+	store_pending();
```

Comment:
Would it make sense to check whether `export_func` actually is `settings_save_one` before calling store_pending? If settings_export is reintroduced, this will currently store all data whenever someone tries to dump it to a log or something

Reply:
Yes, I think, that would make sense.